samples/snippets/csharp/getting-started/console-linq/extensions.cs

```diff
@@ -27,9 +27,9 @@ public static class Extensions
             var firstIter = first.GetEnumerator();
             var secondIter = second.GetEnumerator();
 
-            while (firstIter.MoveNext() && secondIter.MoveNext())
+            while ((firstIter is not null) && firstIter.MoveNext() && secondIter.MoveNext())
```

Comment:
This is interesting. I find it odd that `firstIter` could be null, but `secondIter` can't. At any rate, I think this might be simpler to read:

```suggestion
            while (((firstIter?.MoveNext() == true) && secondIter.MoveNext())
```